server/controllers/finance/account.js

```diff
@@ -0,0 +1,89 @@
+var db = require('../../lib/db');
+
+function create (req, res, next){
+  var record = req.body;
+  var createAccountQuery = 'INSERT INTO account SET ?';
+  
+  db.exec(createAccountQuery, [record])
+  .then(function (result){
+      res.status(201).json({id : result.insertId});
+  })
+  .catch(next)
+  .done();
+}
+
+function update (req, res, next){
+  var queryData = req.body;
+  var accountId = req.params.id;
+  var updateAccountQuery = 'UPDATE account SET ? WHERE id = ?';
+
+  db.exec(updateAccountQuery, [queryData, accountId])
+  .then(function (result){
+    return handleFetchAccount(accountId);
+  })
+  .then(function (accounts){
+    var updatedAccount = accounts[0];
+    res.status(200).json(updatedAccount);
+  })
+  .catch(next)
+  .done();
+}
+
+function list (req, res, next){
+  'use strict';
+
+  var sql = 
+    'SELECT a.id, a.account_number, a.account_txt, a.locked FROM account AS a';
+
+  if(req.query.full === '1'){
+    
+    sql = 
+      'SELECT a.id, a.enterprise_id, a.locked, a.cc_id, a.pc_id, a.created, a.classe, a.is_asset, ' + 
+      'a.reference_id, a.is_brut_link, a.is_used_budget, a.is_charge, a.account_number, ' +
+      'a.account_txt, a.parent, a.account_type_id, a.is_title, at.type FROM account AS a JOIN account_type AS at ON a.account_type_id = at.id';
+  }
+
+  if(req.query.locked === '0'){
+    sql+=' WHERE a.locked = 0';
+  }
+
+  sql += ' ORDER BY a.account_number;';
+
+  db.exec(sql)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+function getAccount (req, res, next){
+  'use strict';
+
+  handleFetchAccount(req.params.id)
+   .then(function (rows) {
+      if(rows.length === 0){
```

Comment:
This should use the error codes defined for our API.  [Here](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/categorised/employees.js#L162) is an example of how to use those codes in the employees API.

It is important to send a full error object back so that the client can properly translate the errors coming from the server/database.  It will also include debugging information for us developers in the future.